### PR TITLE
Refactoring/adjustments

### PR DESCRIPTION
**Pull request by goatherd**
Various minor fixes/ reversions:
- no composer.lock
- adjusted travis.yml
- fixed utf8 issue in class comments
- re-enabled testsuite (using new binary)
- made standard name obligatory for binary selecting (using directory name by default)
- adjusted readme to suggest latest PHP only
- reverted phpcs version range

### CODE REVIEW

**Comment by goatherd** [comment]
@reneoelke for the next steps phpcs2 is required and it is currently used by the master branch so I reverted your changes for that.

Major benefits are:
- support for custom reports
- support for custom CLI arguments
- code cleanup simplifies coding against the library
- most current contributions go against phpcs2

Otherwise there is no difference in just running the tests with the default reporting.

An other change reverted is the `composer.lock`. We opted against those last year and I saw no reasoning to add it here.

**Comment by goatherd** [comment]
@reneoelke experimentally I tested against hhvm which currently fails. If it is trivial to fix I will keep this otherwise it will be dropped for now.

The inclusion of hashbanged phpcs from the binary may be an issue. Will test this later on.

**Comment by goatherd** [comment]
@reneoelke the issue lies with [`tempnam()`](https://github.com/facebook/hhvm/issues/2336) and is resolved in a [pull-request](https://github.com/squizlabs/PHP_CodeSniffer/pull/223).

I will temporarily mark hhvm invalid until this is fixed.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /vendor
 /phpunit.xml
+/composer.lock
```
Comment (reneoelke) [inline .gitignore]: What is the reason for removing the composer.lock file? It is official recommended to save this in git (see https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file "Commit your application's composer.lock (along with composer.json) into version control.").

File: Sniffs/Deprecated/FunctionsSniff.php [hunk 2]
```diff
@@ -20,7 +20,7 @@
  * PHP 5.5.
  *
  * @package Php54to55
- * @author Ren� Oelke <rene.oelke@foobugs.com>
+ * @author René Oelke <rene.oelke@foobugs.com>
```
Comment (reneoelke) [inline Sniffs/Deprecated/FunctionsSniff.php:23]: Sorry, wrong encoding in the project settings of PhpStorm. I changed it.

File: Tests/bootstrap.php [hunk 3]
```diff
@@ -1,4 +1,9 @@
 <?php
 
+use Php54to55\Tests\AbstractTestCase;
+
 require __DIR__ . '/../Sniffs/bootstrap.php';
 require __DIR__ . '/AbstractTestCase.php';
+
+// set binary for integration tests
+AbstractTestCase::$phpcsBinary = dirname(__DIR__) . '/bin/snout';
```
Comment (reneoelke) [inline Tests/bootstrap.php:9]: Perhaps better using `__DIR__ . '/../bin/snout'`?

File: bin/snout [hunk 4]
```diff
@@ -18,9 +23,8 @@ $_SERVER['argc'] = $argc;
 
 $phpcsFilename = realpath($rootPath . '/vendor/bin/phpcs');
 if (is_file($phpcsFilename)) {
-    $result = ob_start(function ($buffer) {
```
Comment (reneoelke) [inline bin/snout:21]: What where the reasons for removing that? Please have a look at `vendor/squizlabs/php_codesniffer/scripts/phpcs`. All that code will be lost (incl. PHP_Timer::start(); $phpcs->checkRequirements(); $phpcs->process(); etc) if not calling this script.

File: composer.json [hunk 5]
```diff
@@ -36,7 +34,7 @@
         }
     ],
     "require": {
-        "squizlabs/php_codesniffer": "1.5.3"
+        "squizlabs/php_codesniffer": "~2@dev"
```
Comment (reneoelke) [inline composer.json]: For now it's okay to use 1.5.3. We want to recommend our application for public use in production environments. So its better to use stable versions.

**Comment by reneoelke** [comment]
1. Please don't use unstable (unreleased) phpcs2. For now its better to use 1.x. For the next steps we could switch to 2.x but in a special branch. Now we want to have a stable refactored master as development base for the upcoming Snout application.
2. Please don't test anything with hhvm here. It costs too much time that is better to use for completing the application.

File: .gitignore (hunk 1 above)
Comment (goatherd) [inline .gitignore]: There was no need for it and we opted against. It is recommended only for non-libraries. As we did not discuss this any further I reverted your change.

File: Tests/bootstrap.php (hunk 3 above)
Comment (goatherd) [inline Tests/bootstrap.php:9]: will be dropped with unit test refactoring.

File: bin/snout (hunk 4 above)
Comment (goatherd) [inline bin/snout:21]: phpcs2 has a cleaner interface for CLI and reporting.
It is simple to adjust your statement to properly use the 1.5 interface with minor adjustments.

**Comment by goatherd** [comment]
@reneoelke adjusted to use stable phpcs 1.5.3 using composer.lock updating to `~1.5`.

With this the pull request is completed. Would you like to review it again?

**Pull request merged**